lib/ApiClient/AbstractCoreApiClient.php

```diff
@@ -217,29 +183,16 @@ protected function buildJsonParameters(array $required, array $optional)
 
     /**
      * Generates the cache key based on the class name, request method, uri and options.
-     *
-     * @param string $method
-     * @param string $uri
-     * @param array  $options
-     *
-     * @return string
      */
-    private function generateCacheKey($method, $uri, array $options = [])
+    private function generateCacheKey(string $method, string $uri, ?array $options = []): string
```

Comment:
Is there a reason, you make options nullable? default can stay as empty array.
```suggestion
    private function generateCacheKey(string $method, string $uri, array $options = []): string
```